framework/i18n/CLocale.php

```diff
@@ -469,4 +469,31 @@ public function getTerritory($id)
 	{
 		return $this->getLocaleDisplayName($id, 'territories');
 	}
+	
+	/** 
+	 * @return array list of local display names of all territories.
+	 * @since 1.1.17
+	 */
+	public function getTerritories()
+	{
+		return $this->_data['territories'];
+	}
+	
+	/** 
+	 * @return array list of local display names of all countries.
+	 * @since 1.1.17
+	 */
+	public function getCountries()
+	{
+		$countries = array();
+		foreach($this->_data['territories'] as $code => $country)
+		{
+			//zz is an unknown region
+			if(($code != 'zz') && preg_match('/[a-z]{2}/', $code))
```

Comment:
`$this->_data['territories'] ` contains all territories - countries as well as world regions and continents. Region codes contain only digits
```
'001' => 'World',
'002' => 'Africa',
```
meanwhile country codes consist of only 2 letters.
```
'ar' => 'Argentina',
'au' => 'Australia',
```
All we have to do is to filter numerical codes leaving only letter codes, i.e. countries.